language-plutus-core/test/Language/PlutusCore/Gen/Type.hs

```diff
@@ -65,18 +67,26 @@ deriveEnumerable ''Kind
 
 deriveEnumerable ''TypeG
 
+
+-- NOTE: The errors we need to handle in property based testing are
+--       when the generator generates garbage or we encounter an
+--       actual type error in testing.
+
+data GenError
+ =  forall n. Show n => Gen (TypeG n) (Kind ())
```

Comment:
I'd just put `n` as a type parameter rather than use the existential. Then you also don't need the constraint - you'll incur it at the site where you actually try to `Show` the error, which is nice anyway.

Reply:
The benefit of this is that I can pick any `n` I like when throwing an error. I agree the constraint in particular is rather ugly.

I tried adding it as a parameter to `GenError` but then I get
```
Occurs check: cannot construct the infinite type: n ~ S n
```
on line 103 which I believe is caused by an error that occurs under a `TyForallG` being in scope `S n` but the whole type is in scope `n`. I.e., the error may be in a different scope than that of the whole type in question.

I also cannot see how to avoid reintroducing the existential in `ErrorP` as there I don't think it makes sense to make it a type parameter.